### DISCUSSION TITLE
Designing an Interactive Online Classroom: Camera, Screen Share, Muting, and Chat

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#5476] [General]
I'm exploring options for an interactive online classroom project.
In this the teachers can share their camera and screen, initially muting students who can later be unmuted by teachers. Also aiming to include a chat feature.

Can someone please provide some recommendations for an effective solution to bring this concept to life?

Thank you

**Comment by lastpeony** [comment]
I suggest exploring the multitrack conference sample within the AMS SDKs.
https://antmedia.io/webrtc-samples/
For video and audio transmission, use Ant Media Server. To implement other online classroom features, utilizing websockets in your application is advisable. Socket.io is a favorable choice due to its user-friendly room feature. For chat you can also use data channels offered by AMS but i would prefer socket.io. 

Your application's server should establish communication with AMS during conference room setup, deletion, and related actions via REST. Remember to review the api calls provided at https://antmedia.io/rest/, particularly the "conference-rooms" route.
